tape/src/main/java/com/squareup/tape/QueueFile.java

```diff
@@ -171,6 +171,13 @@ private void readHeader() throws IOException {
     last = readElement(lastOffset);
   }
 
+  /** Reads the header without any file side effects. */
+  private void readHeaderWithoutFileSideEffects() throws IOException {
+    final long currentFilePointer = raf.getFilePointer();
+    readHeader();
```

Comment:
this still has side effects as it updates member variables, but it should not change any value, right?

Reply:
Yes but I want those member variables to be up to date with what the file is reporting (source of truth). That's why we call it here.